Contentful.AspNetCore/TagHelpers/ContentfulImageTagHelper.cs

```diff
@@ -130,8 +137,107 @@ public ContentfulImageTagHelper(IContentfulClient client)
                 queryBuilder.UseProgressiveJpg();
             }
 
+            return $"{Url}{queryBuilder.Build()}";
+        }
+    }
+
+    /// <summary>
+    /// TagHelper to create an img tag for a Contentful asset.
+    /// </summary>
+    [RestrictChildren("contentful-source")]
+    public class ContentfulImageTagHelper : ImageTagHelperBase
+    {
+        /// <summary>
+        /// Creates a new instance of ContentfulImageTagHelper.
+        /// </summary>
+        /// <param name="client">The IContentfulClient used to retrieve the asset.</param>
+        public ContentfulImageTagHelper(IContentfulClient client)
+        {
+            _client = client;
+        }
+
+        /// <summary>
+        /// Executes the taghelper.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="output"></param>
+        /// <returns></returns>
+        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            var sources = new List<string>();
+            context.Items.Add("sources", sources);
+            context.Items.Add("defaults", this);
+
+            await output.GetChildContentAsync();
+
             output.TagName = "img";
-            output.Attributes.Add("src", Url + queryBuilder.Build());
+            output.Attributes.Add("src", await BuildUrl());
+
+            if (sources.Any())
+            {
+                output.Attributes.Add("srcset", string.Join(", ", sources));
+            }
+        }
+    }
+
+    /// <summary>
+    /// Taghelper that represents a single source in a source set.
+    /// </summary>
+    [HtmlTargetElement(ParentTag = "contentful-image")]
+    public class ContentfulSource : ImageTagHelperBase
+    {
+        /// <summary>
+        /// The size specification for this source.
+        /// </summary>
+        public string Size { get; set; }
+
+        /// <summary>
+        /// Creates a new instance of ContentfulSource.
+        /// </summary>
+        /// <param name="client">The IContentfulClient used to retrieve the asset.</param>
+        public ContentfulSource(IContentfulClient client)
+        {
+            _client = client;
+        }
+
+        /// <summary>
+        /// Executes the taghelper.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="output"></param>
+        /// <returns></returns>
+        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            var defaults = context.Items["defaults"] as ContentfulImageTagHelper;
+            SetDefaults(defaults);
+            var sources = context.Items["sources"] as List<string>;
+            var url = await BuildUrl();
+
+            if (string.IsNullOrEmpty(Size) && (Width > 0 || Height > 0))
+            {
+                var defaultSize = Width > 0 ? Width : Height;
```

Comment:
defaultSize is not used here. I'm not sure if the specification even supports "h" for image heights. If not this might need to be restricted to width only.